flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/checkpointing/SingleCheckpointBarrierHandler.java

```diff
@@ -255,7 +255,6 @@ public void processBarrierAnnouncement(
             CheckpointBarrier announcedBarrier, int sequenceNumber, InputChannelInfo channelInfo)
             throws IOException {
         if (checkNewCheckpoint(announcedBarrier)) {
-            firstBarrierArrivalTime = getClock().relativeTimeNanos();
             if (alternating) {
```

Comment:
Shouldn't we register the Alignment timer on the first barrier as well If there were no announcements for that barrier? I think we could move this block into the `checkNewCheckpoint` method.

Reply:
Good point. I believe it indeed makes sense to move it to `checkNewCheckpoint`

Reply:
As I understand, this will be only partially working anyway: if we received the first barrier from localChannel without announcement and the timeout happened, then only barriers that have the announcements can be prioritized but if the other channel is local there will be no effect.
So how do you think it is problem or not if we can prioritize the barriers only for the remote channel(which has an announcement)?

Reply:
I think that's a valid yet, unrelated issue. It's the same independent if we first receive an announcement or a barrier for a given checkpoint.

Reply:
cc @pnowojski